common/shell/shell_platform.h

```diff
@@ -0,0 +1,91 @@
+#ifndef SHELL_PLATFORM_H
+#define SHELL_PLATFORM_H
+
+/* Declare Command Enable */
+#define COMMAND_GPIO 1
+#define COMMAND_SENSOR 0
+#define COMMAND_PORT 0
+
+/* Declare Common */
+#define sensor_name_to_num(x) #x,
+#define GET_BIT_VAL(val, n) ( (val&BIT(n))>>(n) )
+
+/* Declare GPIO */
+#if COMMAND_GPIO == 1
```

Comment:
Is there currently a need to have the #if COMMAND_GPIO, COMMAND_SENSOR and PINMASK_RESERVE_CHECK throughout the code?  Lots of #ifs specified which might be better served by if/else statements.

Reply:
The reason why I use these macros is that it could make code working as lot of modules that developers can easily remove or add any type of command based on specific section. Once every required commands are ready, it could be removed.

Reply:
After some internal discussion we'd like to try and factor these out as much as possible.

One internal comment from another engineer:

> I'm not in favor of using #ifdefs for this as it means when the code is compiled, code paths are altered in compile state and not during execution, which makes debug harder when trying to follow the source path.